### PR TITLE
Add a regression test for issue-69323

### LINKED ISSUE #69323: Support for partial existentials in TAIT
[[playground]](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f1faf43935fce3efb6f8390c574ee41c) ([Tracking issue](https://github.com/rust-lang/rust/issues/63063#issue-473710023))

With RPIT, it is possible to create a "partial existential":

```rust
fn test1<A: Iterator<Item=&'static str>>(x: A)
    -> Chain<A, impl Iterator<Item=&'static str>>
{
    x.chain(once(","))
}
```

With TAIT, however, this currently produces an error:

```rust
type Chained<A> = Chain<A, impl Iterator<Item=&'static str>>;
fn test2<A: Iterator<Item=&'static str>>(x: A) -> Chained<A> {
    x.chain(once(","))
}
```

This should probably be possible, both for concept parity for factoring unnamed RPIT out into named TAIT, and because this "partial existential" allows parameterizing partial existentials over input types in a way that allows input-type-dependent trait implementation.

### PR DESCRIPTION
**Pull request by JohnTitor**
Closes #69323
r? @jackh726

### CODE REVIEW

File: src/test/ui/type-alias-impl-trait/issue-69323.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// check-pass
+
+// revisions: min_trait full_trait
```
Comment (jackh726) [inline src/test/ui/type-alias-impl-trait/issue-69323.rs:3]: This is a good idea. But let's just do full/min.

**Review by jackh726 — APPROVED** [review]
r=me with nit

File: src/test/ui/type-alias-impl-trait/issue-69323.rs (hunk 1 above)
Comment (JohnTitor) [reply src/test/ui/type-alias-impl-trait/issue-69323.rs]: Other tests use `min/full_tait` and I picked them up (actually, I typoed though). Anyway fixed.

**Comment by JohnTitor** [comment]
Checked it passes on my local this time.
@bors r=jackh726 rollup

**Pull request merged**